cmd/runAsNonRoot.go

```diff
@@ -46,15 +47,65 @@ func checkRunAsNonRoot(container ContainerV1, result *Result) {
 	return
 }
 
+// Checks the PSC for RANR
+
+func checkRunAsNonRootPSC(podSpec PodSpecV1, result *Result) {
+	if reason := result.Labels["audit.kubernetes.io/allow-run-as-root"]; reason != "" {
+		if podSpec.SecurityContext == nil || podSpec.SecurityContext.RunAsNonRoot == nil || *podSpec.SecurityContext.RunAsNonRoot == false {
+			occ := Occurrence{
+				podHost:  podSpec.Hostname,
+				id:       ErrorRunAsNonRootFalseAllowed,
+				kind:     Warn,
+				message:  "Allowed setting RunAsNonRoot to false",
+				metadata: Metadata{"Reason": prettifyReason(reason)},
+			}
+			result.Occurrences = append(result.Occurrences, occ)
+		} else {
+			occ := Occurrence{
+				podHost:  podSpec.Hostname,
+				id:       ErrorMisconfiguredKubeauditAllow,
+				kind:     Warn,
+				message:  "Allowed setting RunAsNonRoot to false, but it is set to true",
+				metadata: Metadata{"Reason": prettifyReason(reason)},
+			}
+			result.Occurrences = append(result.Occurrences, occ)
+		}
+	} else if podSpec.SecurityContext == nil || podSpec.SecurityContext.RunAsNonRoot == nil {
+		occ := Occurrence{
+			podHost: podSpec.Hostname,
+			id:      ErrorRunAsNonRootNil,
+			kind:    Error,
+			message: "RunAsNonRoot is not set, which results in root user being allowed!",
+		}
+		result.Occurrences = append(result.Occurrences, occ)
+	} else if *podSpec.SecurityContext.RunAsNonRoot == false {
+		occ := Occurrence{
+			podHost: podSpec.Hostname,
+			id:      ErrorRunAsNonRootFalse,
+			kind:    Error,
+			message: "RunAsNonRoot is set to false (root user allowed), please set to true!",
+		}
+		result.Occurrences = append(result.Occurrences, occ)
+	}
+	return
+}
+
 func auditRunAsNonRoot(resource Resource) (results []Result) {
+	// get PodSpec for PSC
+	podSpec := getPodSpecs(resource)
 	for _, container := range getContainers(resource) {
 		result, err := newResultFromResource(resource)
 		if err != nil {
 			log.Error(err)
 			return
 		}
 
-		checkRunAsNonRoot(container, result)
+		// check if Container Security Context is defined properly, else audit the Pod Security Context
+		if (container.SecurityContext == nil && podSpec.SecurityContext != nil) || ((container.SecurityContext != nil && container.SecurityContext.RunAsNonRoot == nil) && (podSpec.SecurityContext != nil && podSpec.SecurityContext.RunAsNonRoot != nil)) {
```

Comment:
I think it would be more clear if we just check
if isDefinedProperly(container) {
   checkRunAsRoot(container)
} else {
   ....
}

and ur conditional statements are a bit long, it would be nice if you can break it into multiple lines.

Reply:
I agree with this suggestion, should create a new function to check PSC defined CSC not defined, would change this up